### ISSUE TITLE
Slider broken on new themes when ThumbEnabled is False

### ISSUE THREAD

**Issue by ewpatton** [mit-cml/appinventor-sources#1119]
[From the forum](https://groups.google.com/d/msg/mitappinventortest/5XaR6x9uK20/5sT6SYxdEQAJ): When a new theme other than "Classic" is used and ThumbEnabled is false.

When true:
![image](https://user-images.githubusercontent.com/1331253/34725354-a40885f8-f51e-11e7-9fb7-972cda25cc66.png)

When false:
![image](https://user-images.githubusercontent.com/1331253/34725430-e4b57174-f51e-11e7-8755-fdb82beb1ba1.png)

**Comment by conorshipp** [comment]
I think this might be the same or a similar issue to what Thunkable has.

https://community.thunkable.com/t/slider-thumbposition-bug/288/

**Comment by ewpatton** [comment]
It seems like people tend to do this to use the Slider as a progress bar. It might make sense to introduce a ProgressBar component instead that uses the built-in Android ProgressBar class, which will draw as expected.

**Comment by NmdOfficial** [comment]
Hello Evan,
I have the issue in makeroid fixed since month with this little code:

```
public void ThumbEnabled(boolean enabled) {
    thumbEnabled = enabled;
    seekbar.setOnTouchListener(new OnTouchListener() {
        @Override
        public boolean onTouch(View view, MotionEvent motionEvent) {
            if (thumbEnabled) {
                return false;
            } else {
                return true;	
            }
        }
    });

  }
```
I hope that helps. And this solution looks for me like the best way.
Btw the issue was not introduced with the themes update.
It was much longer before. In thunkable as example is the bug more then a year.

**Comment by ewpatton** [comment]
@NmdOfficial I don't know that this is the same issue. My concern is with the visual glitch, not the actual behavior of the slider. My guess is that if we could fix the issue with the space where the thumb used to be (by making it 0 width), one would no longer get touch updates and thus it would be effectively non-interactive.

**Comment by NmdOfficial** [comment]
It is the same issue. Because when we have startet makeroid we had the same problem.
Then I have fixed it with the above code and everything works since it very good without any problem.

Maybe you can try the code above local. Then you will see if you can solve the AI2 issue with it or not :)

**Comment by kkashi01** [comment]
We don't use that code and do not have the visual issue. The code clearly
shows updates to touch event

On Jan 10, 2018 2:37 PM, "Next Mobile Development" <notifications@github.com>
wrote:

It is the same issue. Because when we have startet makeroid we had the same
problem.
Then I have fixed it with the above code and everything works since it very
good without any problem.

Maybe you can try the code above local. Then you will see if you can solve
the AI2 issue with it or not :)

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/mit-cml/appinventor-sources/issues/1119#issuecomment-356712953>,
or mute the thread

**Comment by NmdOfficial** [comment]
@kkashi01 Which code do you use for the  thumb enabled property?

**Comment by kkashi01** [comment]
I just tried and disabling it causes same issue. Will update code to see result

**Comment by josmas** [comment]
This only affects API level 21+ and there's a method to deal with it explicitly. Please review #1197 if you have a chance. Thanks!

**Comment by ewpatton** [comment]
Closed with companion release 2.47
